### ISSUE TITLE
plugin memory leak on 5.1.0-3

### ISSUE THREAD

**Issue by deric** [collectd/collectd#1232]
I have an instance of collectd running on Debian Wheezy which currently occupies over 16GB in memory. There are several plugins running I'm not sure which one is responsible for the leak, most probable candidate is the `exec` plugin which is running simple bash scripts. Is there a way how to check this?
```
root     22128  0.9 49.3 16482520 16237540 ?   SLsl Jul05 621:24 /usr/sbin/collectd -C /etc/collectd/collectd.conf -P /var/run/collectd.pid
nobody   22136  0.0  0.0  17272  9384 ?        S    Jul05  58:41  \_ bash /usr/share/collectd/pct.bash
nobody   22137  0.1  0.0  17276  9400 ?        S    Jul05 105:21  \_ bash /usr/share/collectd/frontend.bash
``` 
there are huge amounts of memory allocated under the main process (heap 2.8GB):
```
6028:   /usr/sbin/collectd -C /etc/collectd/collectd.conf -P /var/run/collectd.pid
0000000000400000    136K r-x--  /usr/sbin/collectd
0000000000621000      4K r----  /usr/sbin/collectd
0000000000622000      4K rw---  /usr/sbin/collectd
00000000019a2000   1936K rw---    [ anon ]
0000000001b86000 2808268K rw---    [ anon ]
00007fdd48000000  31100K rw---    [ anon ]
00007fdd49e5f000  34436K -----    [ anon ]
00007fdd4c000000   2212K rw---    [ anon ]
00007fdd4c229000  63324K -----    [ anon ]
00007fdd50000000  56860K rw---    [ anon ]
00007fdd53787000   8676K -----    [ anon ]
00007fdd54000000  45716K rw---    [ anon ]
00007fdd56ca5000  19820K -----    [ anon ]
00007fdd58000000  65532K rw---    [ anon ]
...
```
The highest amounts of memory are marked as anonymous which doesn't help much. I'm not sure about these calls, is it ok?
```
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe0c7173000
read(16, "32454 (nginx) S 9942 9942 9942 0"..., 1024) = 235
read(16, "", 1024)                      = 0
close(16)                               = 0
munmap(0x7fe0c7173000, 4096)            = 0
open("/proc/32454/task", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = 16
getdents(16, /* 3 entries */, 32768)    = 80
getdents(16, /* 0 entries */, 32768)    = 0
close(16)                               = 0
open("/proc/32454/status", O_RDONLY)    = 16
fstat(16, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe0c7173000
read(16, "Name:\tnginx\nState:\tS (sleeping)\n"..., 1024) = 777
read(16, "", 1024)                      = 0
close(16)                               = 0
munmap(0x7fe0c7173000, 4096)            = 0
open("/proc/32454/io", O_RDONLY)        = 16
fstat(16, {st_mode=S_IFREG|0400, st_size=0, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe0c7173000
read(16, "rchar: 7516569\nwchar: 10708190\ns"..., 1024) = 128
read(16, "", 1024)                      = 0
close(16)                               = 0
munmap(0x7fe0c7173000, 4096)            = 0
open("/proc/32454/cmdline", O_RDONLY)   = 16
read(16, "nginx: worker process", 4096) = 21
read(16, "", 4075)                      = 0
close(16)  
```
is there an easy way how to find leaking plugin?

**Comment by mfournier** [comment]
I usually investigate this sort of issue by setting a high read interval (such as `Interval 0.5`) and running collectd under valgrind for a few minutes: `valgrind --leak-check=full /usr/sbin/collectd -f`. If you install the `collectd-dbg` package, you'll get more detailed output.

Also, 5.1 isn't supported anymore, so even if there's a leak, there won't be any release in the 5.1 serie :-(

Just to be sure, you might want to check if the problem is still there in current releases by installing a nightly build: https://github.com/collectd/collectd-ci/blob/master/README.md

**Comment by rubenk** [comment]
We fixed a lot of memory leaks in 5.5.1, so it is possible this is already fixed.

I'm going to close this issue, please reopen if you can reproduce this with a recent Collectd version.
